### PR TITLE
CSP-1030 Implements Get leaving reasons endpoint

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AANHub.Domain/Entities/LeavingReason.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace SFA.DAS.AANHub.Domain.Entities;
+
+[ExcludeFromCodeCoverage]
+public class LeavingReason
+{
+    public int Id { get; set; }
+    public string Category { get; set; } = string.Empty;
+    public string Description { get; set; } = string.Empty;
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Domain/Entities/LeavingReason.cs:10]: This should be initialised with null negation.... `= null!;`

File: src/SFA.DAS.AANHub.Data/Configuration/LeavingReasonConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System.Diagnostics.CodeAnalysis;
+
+namespace SFA.DAS.AANHub.Data.Configuration;
+
+[ExcludeFromCodeCoverage]
+public class LeavingReasonConfiguration : IEntityTypeConfiguration<LeavingReason>
+{
+    public void Configure(EntityTypeBuilder<LeavingReason> builder)
+    {
+        builder.ToTable("LeavingReason");
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Data/Configuration/LeavingReasonConfiguration.cs:13]: Could use `nameof (LeavingReason)`

File: src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+public class LeavingReasonProcessed
+{
+    public int Id { get; set; }
+    public string Description { get; set; } = string.Empty;
+    public int Ordering { get; set; }
+}
+
+public record LeavingReasonModel(int Id, string Category, string Description, int Ordering);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs:65]: What is the point of the model when it is same as Entity?

File: src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    });
+                }
+
+                leavingCategories.Add(new LeavingCategory
+                {
+                    Category = category,
+                    LeavingReasons = leavingReasons
+                });
+
+            }
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs:39]: This logic should be in the handler

File: src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
+﻿using SFA.DAS.AANHub.Domain.Entities;
+
+namespace SFA.DAS.AANHub.Application.LeavingReasons.Queries.GetLeavingReasons;
+
+public class GetLeavingReasonsQueryResult
+{
+    public IEnumerable<LeavingReasonModel> LeavingReasons { get; set; } = null!;
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs:7]: Why expose this if it is not being used

File: src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs (hunk 4 above)
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs:39]: The logic can be simplified, using LINQ
`LeavingReasons.GroupBy(l => l.Category).Select(g => new LeavingCategory { Category = g.Key, LeavingReasons = g.ToList() }).ToList();`

**Review by DevenKShah — CHANGES_REQUESTED** [review]

**Review by ManishAgarwalDfE — DISMISSED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**